src/Microsoft.Extensions.Logging.Abstractions/Internal/NullLoggerProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Microsoft.Extensions.Logging.Abstractions.Internal
+{
+    public class NullLoggerProvider : ILoggerProvider
+    {
+        public static NullLoggerProvider Instance => new NullLoggerProvider();
+
+        private NullLoggerProvider()
+        {
+        }
+
+        public ILogger CreateLogger(string categoryName)
+        {
+            return NullLogger.Instance;
```

Comment:
You can cache `NullLoggerProvider` instance inside `NullLoggerProvider`, it's enough to have one singleton.